### PR TITLE
Circle CI v2 yml

### PR DESCRIPTION
**Pull request by siad007**
Added a new circle ci version 2 config file.
Note that i could not tested this but maybe it is still helpful for you.

Related to #3874 .

### CODE REVIEW

**Comment by keradus** [comment]
Hi @siad007 ! Very first - big thanks for support on this topic! Looks promising, indeed!

Sadly, for PHP CS Fixer, the whole point of using CircleCI is to run tests on MacOS (same as we use Travis for Linux and AppVeyor for Windows)

While using docker image instead of build php using brew on host OS directly, we are not testing on MacOS anymore:
```
ker@dus:/tmp λ docker run -it --rm circleci/php:7.2 php -r "print_r(posix_uname()['sysname']);"
Linux
```

Can you update your PR please? thanks !

**Comment by kubawerlos** [comment]
Wasn't the purpose of CircleCI to check if the fixer will work on macOS? Ping @keradus

**Comment by siad007** [comment]
@keradus @kubawerlos added your changes right now and it semms ok. But do you have an idea why the former linux based config produced the test failures?

**Comment by kubawerlos** [comment]
@siad007 in `.circleci/config.yml` in line 22 there are trailing whitespaces.

File: .gitattributes [hunk 1]
```diff
@@ -20,6 +20,7 @@ box.json export-ignore
 check_trailing_spaces.sh export-ignore
 phpmd.xml export-ignore
 phpunit.xml.dist export-ignore
+.circleci export-ignore
```
Comment (kubawerlos) [inline .gitattributes]: Can we move it a little bit up to keep alphabetic order? And also add `/` after the directory name to point it is directory?

**Comment by siad007** [comment]
Ah and btw 8.2 is not supported any longer so i updated to xcode 9.0.

**Comment by keradus** [comment]
Thank you @siad007.

**Pull request merged**